ipvs.py

```diff
@@ -533,18 +559,45 @@ def get_pools(self):
         for msg in self.nlsock.query(req):
             svc_lst = msg.get_attr_list().get('service')
             service = Service.from_attr_list(svc_lst)
-            dests = []
-            out_msg = IpvsMessage(
-                'get_dest', flags=netlink.MessageFlags.MATCH_ROOT_REQUEST,
-                attr_list=IpvsCmdAttrList(service=svc_lst)
-            )
-            for dst_msg in self.nlsock.query(out_msg):
-                dst_lst = dst_msg.get_attr_list().get('dest')
-                dests.append(Dest.from_attr_list(dst_lst, svc_lst.get('af')))
-
+            dests = self.get_dests(svc_lst)
             pools.append(Pool.from_args(
                 service=service,
                 dests=dests
             ))
 
         return pools
+
+    def get_pool(self, svc_lst):
+        s = self.get_service(svc_lst)
+        if s is None:
+            return None
+        dests = self.get_dests(s.to_attr_list())
+        return Pool.from_args(service=s, dests=dests)
+
+    def get_service(self, svc_lst):
+        out_msg = IpvsMessage(
+            'get_service', flags=netlink.MessageFlags.REQUEST,
+            attr_list=IpvsCmdAttrList(service=svc_lst))
+        try:
+            res = self.nlsock.query(out_msg)
+            svc_lst = res[0].get_attr_list().get('service')
+            return Service.from_attr_list(svc_lst)
+        except RuntimeError as e:
```

Comment:
I will try to document today how to run test locally so it is faster to iterate and does not go through the travis overhead.